File: services/ml-service/app/schemas/prediccion.py
from datetime import datetime
from pydantic import BaseModel


class PrediccionRunRequest(BaseModel):
    device_id: str | None = None


class PrediccionResponse(BaseModel):
    id: int
    device_id: str
    model_version: str
    prediction_timestamp: datetime
    failure_probability: float
    remaining_useful_life_days: int
    risk_level: str
    created_at: datetime

    model_config = {"from_attributes": True}


class PrediccionDetail(PrediccionResponse):
    feature_snapshot: dict | None = None


class PrediccionListResponse(BaseModel):
    items: list[PrediccionResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/prediccion.ts
export interface Prediccion {
  id: number;
  device_id: string;
  model_version: string;
  prediction_timestamp: string;
  failure_probability: number;
  remaining_useful_life_days: number;
  risk_level: 'alta' | 'media' | 'baja';
  created_at: string;
}

export interface PrediccionDetail extends Prediccion {
  feature_snapshot: Record<string, number> | null;
}

export interface PrediccionListResponse {
  items: Prediccion[];
  total: number;
  page: number;
  page_size: number;
}

export interface Alerta {
  id: number;
  device_id: string;
  prediccion_id: number;
  nivel_riesgo: 'alta' | 'media' | 'baja';
  descripcion: string | null;
  estado: string;
  created_at: string;
}

export interface AlertaListResponse {
  items: Alerta[];
  total: number;
  page: number;
  page_size: number;
}
